CI: TiKV test randomly failed

https://app.travis-ci.com/github/juicedata/juicefs/builds/248241159
```
2022/03/22 11:34:48.150389 juicefs[23673] <WARNING>: update stats: read-only file system [sql.go:614]
2022/03/22 11:34:48.316155 juicefs[23673] <INFO>: chunk 4130:0 was changed 4800 -> 2448 [tkv.go:1833]
2022/03/22 11:34:48.316215 juicefs[23673] <WARNING>: compact 4130:0 failed: invalid argument [tkv.go:1848]
2022/03/22 11:34:48.316564 juicefs[23673] <INFO>: compacted chunk 1330 was not used [tkv.go:1854]
2022/03/22 11:34:48.316612 juicefs[23673] <INFO>: compaction for 4130:0 is wasted, delete slice 1330 (20000000 bytes) [tkv.go:1861]
    redis_test.go:867: inode 4130 should be compacted, but have 102 slices
--- FAIL: TestTiKVClient (17.51s)
```

--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -2333,15 +2333,18 @@ func (r *redisMeta) compactChunk(inode Ino, indx uint32, force bool) {
 				r.deleteSlice(s.chunkid, s.size)
 			}
 		}
-		if r.rdb.LLen(ctx, r.chunkKey(inode, indx)).Val() > 5 {
-			go func() {
-				// wait for the current compaction to finish
-				time.Sleep(time.Millisecond * 10)
-				r.compactChunk(inode, indx, force)
-			}()
-		}
 	} else {
 		logger.Warnf("compact %s: %s", key, errno)
+	}
+
+	if force {
+		r.compactChunk(inode, indx, force)
+	} else {
+		go func() {
+			// wait for the current compaction to finish
+			time.Sleep(time.Millisecond * 10)
+			r.compactChunk(inode, indx, force)
+		}()
 	}
 }
 

--- a/pkg/meta/sql.go
+++ b/pkg/meta/sql.go
@@ -2123,11 +2123,16 @@ func (m *dbMeta) compactChunk(inode Ino, indx uint32, force bool) {
 	} else {
 		logger.Warnf("compact %d %d: %s", inode, indx, err)
 	}
-	go func() {
-		// wait for the current compaction to finish
-		time.Sleep(time.Millisecond * 10)
+
+	if force {
 		m.compactChunk(inode, indx, force)
-	}()
+	} else {
+		go func() {
+			// wait for the current compaction to finish
+			time.Sleep(time.Millisecond * 10)
+			m.compactChunk(inode, indx, force)
+		}()
+	}
 }
 
 func dup(b []byte) []byte {

--- a/pkg/meta/tkv.go
+++ b/pkg/meta/tkv.go
@@ -1875,11 +1875,16 @@ func (m *kvMeta) compactChunk(inode Ino, indx uint32, force bool) {
 	} else {
 		logger.Warnf("compact %d %d: %s", inode, indx, err)
 	}
-	go func() {
-		// wait for the current compaction to finish
-		time.Sleep(time.Millisecond * 10)
+
+	if force {
 		m.compactChunk(inode, indx, force)
-	}()
+	} else {
+		go func() {
+			// wait for the current compaction to finish
+			time.Sleep(time.Millisecond * 10)
+			m.compactChunk(inode, indx, force)
+		}()
+	}
 }
 
 func (r *kvMeta) CompactAll(ctx Context, bar *utils.Bar) syscall.Errno {

